sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -424,7 +419,9 @@ class program_impl {
   }
 
   bool is_cacheable() const {
-    return is_cacheable_with_build_options(BuildOptions) && AllowKernelsCaching;
+    return is_cacheable_with_build_options(BuildOptions) &&
+           is_cacheable_with_build_options(CompileOptions) &&
+           is_cacheable_with_build_options(LinkOptions) && AllowKernelsCaching;
```

Comment:
As `is_cacheable_with_build_options` function accepts compile and link options in addition to build option, I suggest renaming it to `is_cacheable_with_options` to avoid confusion.

Reply:
Done